router/dataplane.go

```diff
@@ -442,105 +457,303 @@ func (d *DataPlane) Run(ctx context.Context) error {
 	d.running = true
 
 	d.initMetrics()
+	d.initComponents()
 
-	read := func(ingressID uint16, rd BatchConn) {
+	for k, v := range d.bfdSessions {
+		go func(ifID uint16, c bfdSession) {
+			defer log.HandlePanic()
+			if err := c.Run(ctx); err != nil && err != bfd.AlreadyRunning {
+				log.Error("BFD session failed to start", "ifID", ifID, "err", err)
+			}
+		}(k, v)
+	}
+
+	d.mtx.Unlock()
 
-		msgs := conn.NewReadMessages(inputBatchCnt)
-		for _, msg := range msgs {
-			msg.Buffers[0] = make([]byte, bufSize)
+	<-ctx.Done()
+	return nil
+}
+
+func (d *DataPlane) initComponents() {
+	g := &errgroup.Group{}
+	d.interfaces = make(map[uint16]NetworkInterface)
+	d.interfaces[0] = NetworkInterface{
+		InterfaceId: 0,
+		Conn:        d.internal,
+	}
+	for id, conn := range d.external {
+		d.interfaces[id] = NetworkInterface{
+			InterfaceId: id,
+			Conn:        conn,
 		}
-		writeMsgs := make(underlayconn.Messages, 1)
-		writeMsgs[0].Buffers = make([][]byte, 1)
+	}
+	//SET DEFAULTS
+	d.procRoutines = uint32(runtime.GOMAXPROCS(0))
```

Comment:
We still need a default value. And writing for example "8" might be good on some machines but not others. So already using "the correct value" makes more sense instead of introducing something that might not be good and will definitely be changed in the next few weeks.